datafusion/core/src/physical_planner.rs

```diff
@@ -256,11 +256,7 @@ fn create_physical_name(e: &Expr, is_first_expr: bool) -> Result<String> {
                         "aggregate expression with filter is not supported"
                     );
                 }
-                if order_by.is_some() {
-                    return exec_err!(
-                        "aggregate expression with order_by is not supported"
-                    );
-                }
+
```

Comment:
The comment above 
```
                // TODO: Add support for filter and order by in AggregateUDF
```

Could be updated to reflect only the filter is not supported